allennlp/commands/evaluate.py

```diff
@@ -196,7 +197,8 @@ def evaluate_from_args(args: argparse.Namespace) -> Dict[str, Any]:
         logger.info("%s: %s", key, metric)
 
     output_file = args.output_file
-    if output_file:
-        with open(output_file, "w") as file:
-            json.dump(metrics, file, indent=4)
+    if not output_file:
+        output_file = os.path.join(args.archive_file.split("model.tar.gz")[0], 'metrics.json')
```

Comment:
This sort of feature I would implement on the command line (or in a script). Something like:
`export MY_DIR=my/path/to/model && allennlp evaluate --archive_file=$MY_DIR/model.tar.gz --output-file=$MY_DIR/metrics.json --other_args=foo`

Reply:
That keeps the interface to evaluate standard meaning fewer merge conflicts. Also, you can use CTRL-R on the command line to quickly recall such commands. So it wouldn't mean more typing either! :)